File: src/main/java/fi/quanfoxes/types/Long.java
package fi.quanfoxes.types;

import fi.quanfoxes.lexer.NumberType;

public class Long extends Number {
    private static final int BYTES = 8;

    public Long() {
        super(NumberType.INT64, 64, "long");
    }

    @Override
    public int getSize() {
        return BYTES;
    }
}

File: Zigzag/Types/Long.cs
public class Long : Number
{
	private const int BYTES = 8;

	public Long() : base(NumberType.INT64, 64, "long") { }

	public override int GetReferenceSize()
	{
		return BYTES;
	}

	public override int GetContentSize()
	{
		return BYTES;
	}
}
